crypto/src/util/mod.rs

```diff
@@ -13,4 +13,66 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use generic_array::{sequence::Split, typenum::U32, GenericArray};
+use hmac::{Hmac, Mac};
+use sha2::Sha512;
+use zeroize::Zeroize;
+
+use crate::key::hdkd::{
+    chain_code::{ChainCode, CHAINCODE_LENGTH},
+    derivable::DerivationError,
+};
+
 pub mod eq;
+
+pub fn new_hmac_sha_512(key: &[u8]) -> Hmac<Sha512> {
+    Hmac::<Sha512>::new_from_slice(key).expect("HMAC can take key of any size")
+}
+
+pub fn to_key_and_chain_code<SecretKey>(
```

Comment:
If this is secp256k1 specific, why is it in the utils, instead of in a module there?

Reply:
The `SecretKey` can be either secp256k1 or VRF key

Reply:
Alrighty.